tests/trx_generator/trx_generator.cpp

```diff
@@ -5,200 +5,294 @@
-      const std::vector<std::string>& accts, int64_t trx_expr, const std::vector<std::string>& private_keys_str_vector,
-      std::string lib_id_str, std::string log_dir, bool stop_on_trx_failed, const std::string& peer_endpoint, unsigned short port) :
-      _provider(peer_endpoint, port), _chain_id(chain_id_in), _handler_acct(handler_acct), _accts(accts),
-      _trx_expiration(trx_expr*1000000), _private_keys_str_vector(private_keys_str_vector),
-      _last_irr_block_id(fc::variant(lib_id_str).as<block_id_type>()), _log_dir(log_dir),
-      _stop_on_trx_failed(stop_on_trx_failed) {
-   }
+   trx_generator_base::trx_generator_base(const trx_generator_base_config& trx_gen_base_config, const provider_base_config& provider_config)
+       : _config(trx_gen_base_config), _provider(provider_config) {}
 
-   void log_first_trx(const std::string& log_dir, const chain::signed_transaction& trx) {
-      std::ostringstream fileName;
-      fileName << log_dir << "/first_trx_" << getpid() << ".txt";
-      std::ofstream out(fileName.str());
+   transfer_trx_generator::transfer_trx_generator(const trx_generator_base_config& trx_gen_base_config, const provider_base_config& provider_config, const accounts_config& accts_config)
+       : trx_generator_base(trx_gen_base_config, provider_config), _accts_config(accts_config) {}
 
-      out << fc::string(trx.id()) << "\n";
-      out.close();
+
+
+
+
+   bool transfer_trx_generator::setup() {
+      const std::string salt = std::to_string(getpid());
+      const uint64_t &period = 20;
+      _nonce_prefix = 0;
+      _nonce = static_cast<uint64_t>(fc::time_point::now().sec_since_epoch()) << 32;
+
+      ilog("Stop Generation (form potential ongoing generation in preparation for starting new generation run).");
+      stop_generation();
+
+      ilog("Create All Initial Transfer Action/Reaction Pairs (acct 1 -> acct 2, acct 2 -> acct 1) between all provided accounts.");
+      create_initial_transfer_actions(salt, period);
+
+      ilog("Create All Initial Transfer Transactions (one for each created action).");
+      create_initial_transfer_transactions(++_nonce_prefix, _nonce);
+
+      ilog("Setup p2p transaction provider");
+
+      ilog("Update each trx to qualify as unique and fresh timestamps, re-sign trx, and send each updated transactions via p2p transaction provider");
+
+      _provider.setup();
+      return true;
    }
 
-   void transfer_trx_generator::push_transaction(p2p_trx_provider& provider, signed_transaction_w_signer& trx, uint64_t& nonce_prefix, uint64_t& nonce, const fc::microseconds& trx_expiration, const chain_id_type& chain_id, const block_id_type& last_irr_block_id) {
-      update_resign_transaction(trx._trx, trx._signer, ++nonce_prefix, nonce, trx_expiration, chain_id, last_irr_block_id);
-      if (_txcount == 0) {
-         log_first_trx(_log_dir, trx._trx);
+   fc::variant trx_generator::json_from_file_or_string(const string& file_or_str, fc::json::parse_type ptype)
+   {
```

Comment:
Brace location is inconsistent here. Aesthetic, but jarring. Feel free to ignore.

Reply:
Addressed: 173fff95